src/gui/editorwidgets/qgsvaluerelationconfigdlg.cpp

```diff
@@ -70,6 +71,23 @@ QVariantMap QgsValueRelationConfigDlg::config()
 void QgsValueRelationConfigDlg::setConfig( const QVariantMap &config )
 {
   QgsVectorLayer *lyr = QgsProject::instance()->mapLayer<QgsVectorLayer *>( config.value( QStringLiteral( "Layer" ) ).toString() );
+  // If layer is null, let's try to match it against name and provider
+  if ( ! lyr )
```

Comment:
Look into QgsVectorLayerRef, and QgsMapLayerRef::resolveWeakly. It's designed to encapsulate all this logic for you.

Reply:
I looked into that, I don't think it fits here, we don't want to check for datasource and provider type, there is a strong use case for layers migrated into a DB where datasource and provided type have changed.

Reply:
Hmm - so we need a "resolveSuperWeakly" method?

Reply:
Maybe, I would probably prefer to have a template based getLayersByName<TYPE> and get the first element out of the results if not empty.